app/src/main/java/lt/vilnius/tvarkau/SettingsActivity.kt

```diff
@@ -0,0 +1,114 @@
+import android.widget.CompoundButton
+import android.widget.Toast
+import com.vinted.extensions.gone
+import com.vinted.extensions.visible
+import kotlinx.android.synthetic.main.settings_activity.*
+import lt.vilnius.tvarkau.ProblemsListActivity.MY_PROBLEMS
+import lt.vilnius.tvarkau.fragments.ReportImportDialogFragment
+import lt.vilnius.tvarkau.utils.SharedPrefsManager
+
+class SettingsActivity : BaseActivity(),
+        CompoundButton.OnCheckedChangeListener,
+        ReportImportDialogFragment.SettingsVilniusSignInListener {
+
+    private lateinit var prefsManager: SharedPrefsManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.settings_activity)
+
+        prefsManager = SharedPrefsManager.getInstance(this)
+        share_contacts_switcher.isChecked = !prefsManager.isUserAnonymous
+        share_contacts_switcher.setOnCheckedChangeListener(this)
+        setUpedit_personal_data(prefsManager.isUserAnonymous)
+        if (prefsManager.userLastReportImport != null) {
+            last_import.visible()
+            last_import.text = "${resources.getString(R.string.last_import)}  ${prefsManager.userLastReportImport}"
+            login_to_vilnius_account.gone()
+        } else {
+            last_import.gone()
+            login_to_vilnius_account.visible()
+        }
+
+        edit_personal_data.setOnClickListener { startProfileEditActivity() }
+        import_reports_from_previous_app.setOnClickListener {
+            val ft = supportFragmentManager.beginTransaction()
+            val reportImportDialog = ReportImportDialogFragment.newInstance(true)
+            reportImportDialog.show(ft, REPORT_IMPORT_DIALOG)
+        }
+    }
+
+    private fun startProfileEditActivity() {
+        Intent(this, ProfileEditActivity::class.java).run {
+            startActivityForResult(this, REQUEST_EDIT_PROFILE)
+        }
+    }
+
+    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
+        if (isChecked) {
+            if (!prefsManager.isUserDetailsSaved) {
+                startProfileEditActivity()
+            } else {
+                setUpedit_personal_data(false)
+                prefsManager.changeUserAnonymityStatus(false)
+            }
+        } else {
+            prefsManager.changeUserAnonymityStatus(true)
+            setUpedit_personal_data(true)
+        }
+    }
+
+    private fun setUpedit_personal_data(isUserAnonymous: Boolean) {
+        if (isUserAnonymous) {
+            edit_personal_data.gone()
+            settings_first_divider.gone()
+        } else {
+            edit_personal_data.visible()
+            settings_first_divider.visible()
+            edit_personal_data.isClickable = true
+            edit_personal_data.setTextColor(getColor(this, R.color.black_87))
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
`?` indicates that property could be null and developer must do a null check before accessing it.
```
val textView : TextView?

//one way
if(textView != null) {
  val text = view.text
}

//another way
val text = textView?.text
```